app_feup/lib/view/Widgets/bug_report_form.dart

```diff
@@ -168,19 +178,47 @@ class BugReportFormState extends State<BugReportForm> {
     );
   }
 
+  Widget consentBox(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.only(),
+      margin: EdgeInsets.only(bottom: 20, top: 0),
+      child: ListTileTheme(
+        contentPadding: EdgeInsets.all(0),
+        child: CheckboxListTile(
+          activeColor: Theme.of(context).primaryColor,
+          title: Text(
+              '''Consinto que a informação seja publicada no GitHub, incluindo o meu contacto pessoal, se fornecido.''',
+              style: Theme.of(context).textTheme.body1,
+              textAlign: TextAlign.left),
+          value: _isConsentGiven,
+          onChanged: (bool newValue) {
+            setState(() {
+              _isConsentGiven = newValue;
+            });
+          },
+          controlAffinity: ListTileControlAffinity.leading,
+        ),
+      ),
+    );
+  }
+
   Widget submitButton(BuildContext context) {
-    return  Container(
+    return Container(
         child: RaisedButton(
       padding: EdgeInsets.symmetric(vertical: 10.0),
-      onPressed: () {
-        if (_formKey.currentState.validate() && !_isButtonTapped) {
-          submitBugReport();
-        }
-      },
+      onPressed: !_isConsentGiven
+          ? null
+          : () {
+              if (_formKey.currentState.validate() && !_isButtonTapped) {
+                submitBugReport();
+              }
+            },
       child: Text(
         'Enviar',
         style: TextStyle(color: Colors.white, fontSize: 20.0),
       ),
+      disabledColor: Theme.of(context).disabledColor,
+      disabledElevation: 1,
```

Comment:
Shouldn't the `isConsentGiven` be passed to the `enabled` field of the button? https://api.flutter.dev/flutter/material/MaterialButton/enabled.html

Reply:
I think that field is not accessible, from what I've seen here: https://api.flutter.dev/flutter/material/RaisedButton-class.html
The button is disabled when onPressed is null